### ISSUE TITLE
CsvCut feature request

### ISSUE THREAD

**Issue by morro91** [wireservice/csvkit#673]
Say I have a 2 column csv with headers "col1" and "col3"

`csvcut -c "col1","col2","col3" filename.csv`

As "col2" doesn't exist it returns an error (as it should based on the code). Can there be an additional attribute, that creates an empty column with that header as well as col1 & col3 which do exist.

Or is there another way this could be done, perhaps through some piping to other csvkit commands, I'm fairly new to it.

The example above is simplified, there is a reason I'm using csvcut :) I'm parsing logs, reducing alot of the columns, but sometimes different columns are dropped from the log if no data exists for that day in that column :( so I'm parsing/formating them into standard versions as part of a batch script

**Comment by jpmckinney** [comment]
You can add an empty column to a file with:

    echo "col2" | csvjoin filename.csv -

But if the file already contains that column header, it will still add it. Does that help though?

**Comment by jpmckinney** [comment]
Oh, but if you then do `csvcut` after the `csvjoin`, the first occurrence of the column (with the data) will be cut, not the second occurrence that is empty.

**Comment by morro91** [comment]
Hmmm, With the echo line you said, I got the following error:

`local variable 'right_row' referenced before assignment`

With the following csv file (just as a test):

```
col1,col3
1,3
1,3
1,3
```

**Comment by jpmckinney** [comment]
You need the latest csvkit:

    pip install --upgrade -e git+git://github.com/wireservice/csvkit.git@master#egg=csvkit

**Comment by morro91** [comment]
Cool, worked good.

I also tried:

`echo "col2","col4" | csvjoin test2.csv - | csvcut -c "col1","col2","col3","col4"`

And it worked well. As I wouldn't know which columns in my CSV would be there or not, I also tried:

`echo "col1","col2","col3","col4" | csvjoin test2.csv - | csvcut -c "col1","col2","col3","col4"`

And as you said, it worked as expected even if columns already in the CSV. Thanks.

For my own future reference:

```
TEST="col1","col2","col3","col4"
echo $TEST | csvjoin test2.csv - | csvcut -c $TEST
```
